File: lib/token.ml
type token_type =
  | Illegal
  | Eof
  | Identifier
  | Integer
  | Comma
  | Semicolon
  | Lparen
  | Rparen
  | Lbrace
  | Rbrace
  (*keywords*)
  | Function
  | Let
  | True
  | False
  | If
  | Else
  | Return
  (*operators*)
  | Assign
  | Plus
  | Minus
  | Asterics
  | Slash
  | Lt
  | Gt
  | Bang
  | Eq
  | Not_eq
[@@deriving show]

type token =
  { token_type : token_type
  ; literal : string
  }
[@@deriving show]

let rec print_tokens = function
  | [] -> ()
  | x :: xs ->
    print_endline @@ show_token x;
    print_tokens xs
;;

File: go_rewrite/token/token.go
package token

const (
	ILLEGAL    = "ILLEGAL"
	EOF        = "EOF"
	IDENTIFIER = "IDENTIFIER"
	INTEGER    = "INTEGER"
	COMMA      = "COMMA"
	SEMICOLON  = "SEMICOLON"
	LPAREN     = "LPAREN"
	RPAREN     = "RPAREN"
	LBRACE     = "LBRACE"
	RBRACE     = "RBRACE"
	//keywords
	FUNCTION = "FUNCTION"
	LET      = "LET"
	TRUE     = "TRUE"
	FALSE    = "FALSE"
	IF       = "IF"
	ELSE     = "ELSE"
	RETURN   = "RETURN"
	//operators
	ASSIGN   = "ASSIGN"
	PLUS     = "PLUS"
	MINUS    = "MINUS"
	ASTERICS = "ASTERICS"
	SLASH    = "SLASH"
	LT       = "LT"
	GT       = "GT"
	BANG     = "BANG"
	EQ       = "EQ"
	NOT_EQ   = "NOT_EQ"
)

type Token struct {
	Token   string
	Literal string
}
